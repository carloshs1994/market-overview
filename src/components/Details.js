import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { IoIosArrowDown, IoIosArrowUp } from 'react-icons/io';
import { getStockDetails, removeDetailsData } from '../redux/details/details';
import PrintDetails from './PrintDetails';

const Details = () => {
  const ticker = useParams();
  const stock = useSelector((state) => state.detailsReducer[0]);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getStockDetails(ticker));
    return () => {
      dispatch(removeDetailsData());
    };
  }, []);
  return (
    <main>
      {(stock)
        ? (
          <>
            <section className="stock">
              <h1>{ticker.symbol}</h1>
              <h2>{stock.name}</h2>
              <p>
                Price:
                {' $'}
                {stock.price}
                {' '}
                {(stock.change <= 0)
                  ? (
                    <span style={{ color: 'rgb(255, 152, 152)' }}>
                      <IoIosArrowDown />
                      {stock.change}
                    </span>
                  )
                  : (
                    <span style={{ color: 'rgb(126, 255, 126)' }}>
                      <IoIosArrowUp />
                      {stock.change}
                    </span>
                  )}
              </p>
            </section>
            <section className="description">
              <p>{`Detailed Stats About ${stock.symbol}`}</p>
            </section>
            <ul className="details">
              <PrintDetails stockDetails={stock} />
            </ul>
          </>
        ) : (<h1>loading</h1>)}
    </main>
  );
};

export default Details;
