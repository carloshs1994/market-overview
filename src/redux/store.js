import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import homeReducer from './home/home';
import detailsReducer from './details/details';

const reducers = combineReducers({
  homeReducer,
  detailsReducer,
});

const store = createStore(
  reducers,
  applyMiddleware(logger, thunk),
);

export default store;
