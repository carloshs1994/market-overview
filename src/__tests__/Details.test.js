import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Details from '../components/Details';
import store from '../redux/store';

test('matches voice search component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Details />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
