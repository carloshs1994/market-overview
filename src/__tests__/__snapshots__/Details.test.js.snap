// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches voice search component snapshot 1`] = `
<main>
  <h1>
    loading
  </h1>
</main>
`;
